solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -1878,7 +1870,6 @@ private int fetchPackets(FastInputStream fis) throws Exception {
           log.debug("Fetched and wrote {} bytes of file: {}", bytesDownloaded, fileName);
           // errorCount is always set to zero after a successful packet
           errorCount = 0;
-          if (bytesDownloaded >= size) return 0;
```

Comment:
I did a git blame and found: SOLR-14299 IndexFetcher doesn't reset count to 0 after the last packet is received